src/Transports.AspNetCore/AuthorizationVisitorBase.GetRecursivelyReferencedFragments.cs

```diff
@@ -0,0 +1,100 @@
+    /// Returns all fragments referenced by the selected operation in the document,
+    /// excluding ones that would be skipped by the @skip or @include directives.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="SkipNode(ASTNode, ValidationContext)"/> is used to determine if the node should be skipped or not.
+    /// </remarks>
+    protected List<GraphQLFragmentDefinition>? GetRecursivelyReferencedFragments(ValidationContext validationContext)
+    {
+        var context = new GetRecursivelyReferencedFragmentsVisitorContext(this, validationContext);
+        var ret = GetRecursivelyReferencedFragmentsVisitor.Instance.VisitAsync(validationContext.Operation, context);
+        if (!ret.IsCompletedSuccessfully) // should always be true unless an exception occurs within SkipNode
+            ret.AsTask().GetAwaiter().GetResult();
+        return context.FragmentDefinitions;
+    }
+
+    private sealed class GetRecursivelyReferencedFragmentsVisitorContext : IASTVisitorContext
+    {
+        public GetRecursivelyReferencedFragmentsVisitorContext(AuthorizationVisitorBase authorizationVisitor, ValidationContext validationContext)
+        {
+            AuthorizationVisitor = authorizationVisitor;
+            FragmentDefinitions = null;
+            ValidationContext = validationContext;
+        }
+
+        public AuthorizationVisitorBase AuthorizationVisitor { get; }
+
+        public CancellationToken CancellationToken => default;
+
+        public ValidationContext ValidationContext { get; }
+
+        public List<GraphQLFragmentDefinition>? FragmentDefinitions { get; set; }
+    }
+
+    private sealed class GetRecursivelyReferencedFragmentsVisitor : ASTVisitor<GetRecursivelyReferencedFragmentsVisitorContext>
+    {
+        private GetRecursivelyReferencedFragmentsVisitor() { }
+
+        public static readonly GetRecursivelyReferencedFragmentsVisitor Instance = new();
+
+        public override ValueTask VisitAsync(ASTNode? node, GetRecursivelyReferencedFragmentsVisitorContext context)
+        {
+            // check if this node should be skipped or not (check @skip and @include directives)
+            if (node == null || !context.AuthorizationVisitor.SkipNode(node, context.ValidationContext))
+                return base.VisitAsync(node, context);
+
+            return default;
+        }
+
+        protected override ValueTask VisitSelectionSetAsync(GraphQLSelectionSet selectionSet, GetRecursivelyReferencedFragmentsVisitorContext context)
+            => VisitAsync(selectionSet.Selections, context);
+
+        protected override ValueTask VisitFieldAsync(GraphQLField field, GetRecursivelyReferencedFragmentsVisitorContext context)
+            => VisitAsync(field.SelectionSet, context);
+
+        protected override ValueTask VisitInlineFragmentAsync(GraphQLInlineFragment inlineFragment, GetRecursivelyReferencedFragmentsVisitorContext context)
+            => VisitAsync(inlineFragment.SelectionSet, context);
```

Comment:
```suggestion
```

FYI. The same may be done for visitor from core repo. Everything will work after removing. The difference is performance because you skip visiting comments, directives, type conditions, etc. Or you may even override more methods to gain perf, for example, `VisitVariableAsync/VisitVariablesDefinitionAsync`

Reply:
I understand.  With the addition of overriding the operation, all other node types should be skipped, I believe.  I'll make a separate PR.

Reply:
#847